src/verification/android.rs

```diff
@@ -73,7 +72,7 @@ impl Verifier {
         &self,
         end_entity: &rustls::Certificate,
         intermediates: &[rustls::Certificate],
-        server_name: &str,
+        server_name: &rustls::ServerName,
```

Comment:
If we could make this method private we could just take a `&str` here and do the conversion to string in the `ServerCertVerifier::verify_server_cert()` method, instead? That would avoid the `unreachable!()` shenanigans, at least...

Reply:
I already made this `fn` private in https://github.com/rustls/rustls-platform-verifier/pull/24 so I think its fine to bring that change in here.

We'd need to pass the `rustls::ServerName` in addition to the `&str`, but otherwise seems OK.

Reply:
> If we could make this method private we could just take a &str here

Also, I think it needs to be `String`. You can get a `&str` from `DnsName` but not from `IpAddr` AFAICT.

Reply:
Pushed an update to accomplish the above. PTAL